VTCareService/src/main/java/edu/vt/cs/vtcare/vtcareservice/models/Provider.java

```diff
@@ -0,0 +1,88 @@
+package edu.vt.cs.vtcare.vtcareservice.models;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.sql.Date;
+import java.text.SimpleDateFormat;
+import java.time.LocalDate;
+import java.util.List;
+
+public class Provider extends User {
+
+    private long providerId;
+    private int yearsOfExperience;
+    private String specialization;
+
+    private List<Appointment> upcomingAppointments;
+    private List<AppointmentSlot> availabilitySchedule;
+
+    @JsonCreator
+    public Provider(@JsonProperty("name") String name,
+                    @JsonProperty("email") String email,
+                    @JsonProperty("password") String password,
+                    @JsonProperty("gender") String gender,
+                    @JsonProperty("dob") String dob,
+                    @JsonProperty("address") String address,
+                    @JsonProperty("contact") String contact,
+                    @JsonProperty("experience") int experience,
+                    @JsonProperty("specialization") String specialization) {
+        this.setName(name);
+        this.setEmail(email);
+        this.setPassword(password);
+        this.setGender(gender);
+        this.setDate(dob);
+        this.setAddress(address);
+        this.setContact(contact);
+        this.yearsOfExperience = experience;
+        this.specialization = specialization;
+    }
+
+    public long getProviderId() {
```

Comment:
If we are making each variable private, and then adding getter/setter methods for it, why don't we just make them public?

Reply:
On the contrary, if we make raw getters and setters like this, it still violates encapsulation in the sense that we give access to it publically, but just in functions. It was just a question, it's debatable, and a preference for particular models like this.

Reply:
It's the industry and literature standard to do it this way, I would again refer you to do more R&D on getters and setters in Java. Will help you understand it better. 
Getters and setters 'are' made only like this. I am sure you are missing a major point here.